user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -97,6 +97,11 @@ define([
     return Encoder.encode("{0}", path);
   };
 
+  FileBrowser.encodeGenericFilePathComponents = function (path) {
```

Comment:
unit tests for` FileBrowser.encodeGenericFilePathComponents `.

The other uses of the function, don't see easily reachable. 

I'm not sure browser.js has a corresponding test class existing or if there is a test framework in place. If the level of effort becomes "unreasonable" I'm fine with spiking unit tests for now.

Reply:
I didn't find an existing "karma" unit testing setup here like I saw and added to [for URLEncoder in common-ui](https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/test/javascript/util/URLEncoder.spec.js) There are configuration files and such I've [found here](https://github.com/pentaho/pentaho-platform-plugin-common-ui/tree/master/impl/client/src/test/config/javascript)... but the only way I could get this stuff to work was by running `mvn test`, I haven't used the framework before.

Reply:
There is no test fwk setup for this area of the code.

Reply:
Follow on task to to look into setting up unit test framework for this project - https://hv-eng.atlassian.net/browse/BACKLOG-40575 has run over it's time. I'll merge and we can look at this another time.